torch/distributed/_tensor/device_mesh.py

```diff
@@ -227,10 +232,24 @@ def __eq__(self, other: object) -> bool:
             return True
         return self.mesh.equal(other.mesh)
 
-    def get_dim_groups(self) -> List[ProcessGroup]:
-        if not hasattr(self, "_dim_groups"):
+    def get_dim_groups(
+        self, mesh_dim: Optional[int] = None
+    ) -> Union[ProcessGroup, List[ProcessGroup]]:
+        if not hasattr(self, "_dim_group_ranks"):
```

Comment:
same as above, if we combine ranks and tag into a tuple, we just use "_dim_groups" here.